Sources/NIO/CircularBuffer.swift

```diff
@@ -89,11 +116,31 @@ public struct CircularBuffer<E>: CustomStringConvertible, AppendableCollection {
     ///
     /// *O(1)*
     public var first: E? {
-        if self.isEmpty {
-            return nil
-        } else {
-            return self.buffer[self.headIdx]
-        }
+        // As self.buffer is filled with `nil` this will correctly return `nil` if the CircularBuffer is empty.
```

Comment:
Is this change worth it? The path of checking `isEmpty` is going to be faster, because it saves us the additional level of indirection into the buffer.

Reply:
@Lukasa not sure I think it is a nice change but I am not strong on it...

Reply:
@weissi for the tie-break!

Reply:
feels to clever to me, I'd leave the old code

Reply:
@weissi achievement unlocked... @weissi called me clever =P 

Let me revert this part